src/service/search/grpc/mod.rs

```diff
@@ -210,3 +224,49 @@ pub async fn search(
 
     Ok(result)
 }
+
+pub fn handle_datafusion_error(err: DataFusionError) -> Error {
+    let err = err.to_string();
+    if err.contains("Schema error: No field named") {
+        let pos = err.find("Schema error: No field named").unwrap();
+        return match get_key_from_error(&err, pos) {
+            Some(key) => Error::ErrorCode(ErrorCodes::SearchFieldNotFound(key)),
+            None => Error::ErrorCode(ErrorCodes::SearchSQLExecuteError(err)),
+        };
+    }
+    if err.contains("parquet not found") {
+        return Error::ErrorCode(ErrorCodes::SearchParquetFileNotFound);
+    }
+    if err.contains("Invalid function ") {
+        let pos = err.find("Invalid function ").unwrap();
+        return match get_key_from_error(&err, pos) {
+            Some(key) => Error::ErrorCode(ErrorCodes::SearchFunctionNotDefined(key)),
+            None => Error::ErrorCode(ErrorCodes::SearchSQLExecuteError(err)),
+        };
+    }
+    if err.contains("Incompatible data types") {
+        let pos = err.find("for field").unwrap();
+        let pos_start = err[pos..].find(' ').unwrap();
+        let pos_end = err[pos + pos_start + 1..].find('.').unwrap();
+        let field = err[pos + pos_start + 1..pos + pos_start + 1 + pos_end].to_string();
+        return Error::ErrorCode(ErrorCodes::SearchFieldHasNoCompatibleDataType(field));
+    }
+    Error::ErrorCode(ErrorCodes::SearchSQLExecuteError(err))
+}
+
+fn get_key_from_error(err: &str, pos: usize) -> Option<String> {
+    for ponct in ['\'', '"'] {
```

Comment:
```suggestion
    for punct in ['\'', '"'] {
```
if you meant ‘punctuation’.

Or
```suggestion
    for quot in ['\'', '"'] {
```